test/functional/ui/decorations_spec.lua

```diff
@@ -2120,4 +2120,49 @@ describe('decorations: virt_text', function()
     ]]}
   end)
 
+  it('XXX', function()
```

Comment:
Needs a better name

Reply:
I'm not even sure if we should even include this test, as it's quite convoluted. What do you think?